changelogs/unreleased/gh-5316-msgpack-object.md

```diff
@@ -0,0 +1,4 @@
+## feature/lua/msgpack
```

Comment:
I'm not quite sure what such 'introspection' methods should do:
- Should `type` return MsgPack type (e.g. uint, int, float, double, map, array) or Lua type (e.g. number, table)?
- May be we should have two methods - `mp_type` (returns MsgPack type) and `type` (returns Lua type)?
- Do we need `is_array` and `is_map` helpers?

Since we don't need introspection for passing through function arguments, which is the main goal of this change, I decided to remove these methods from the API for now. I agree that without them the API would probably be incomplete, but I suggest to add them in a follow-up PR.